# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1375991550 14400
#      Thu Aug 08 15:52:30 2013 -0400
# Node ID 2527bc7200ee29eb343e0917128a184665775e67
# Parent  8b64a1fc3071521d790819049dcca73120ec7039
Only scroll history window if it is positioned at bottom.

* history-dock-widget.cc
(history_dock_widget::append_history): Scroll to bottom if slide bar value
equals maximum.

diff --git a/libgui/src/history-dock-widget.cc b/libgui/src/history-dock-widget.cc
--- a/libgui/src/history-dock-widget.cc
+++ b/libgui/src/history-dock-widget.cc
@@ -23,16 +23,17 @@ along with Octave; see the file COPYING.
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <QApplication>
 #include <QClipboard>
 #include <QVBoxLayout>
 #include <QMenu>
+#include <QScrollBar>
 
 #include "error.h"
 
 #include "cmd-hist.h"
 
 #include "history-dock-widget.h"
 
 history_dock_widget::history_dock_widget (QWidget *p)
@@ -155,18 +156,26 @@ history_dock_widget::set_history (const 
   _history_list_view->scrollToBottom ();
 }
 
 void
 history_dock_widget::append_history (const QString& hist_entry)
 {
   QStringList lst = _history_model->stringList ();
   lst.append (hist_entry);
+
+  QScrollBar *scroll_bar = _history_list_view->verticalScrollBar ();
+
+  bool at_bottom = scroll_bar->maximum () - scroll_bar->value () < 1;
+
   _history_model->setStringList (lst);
-  _history_list_view->scrollToBottom ();
+
+  // Scroll if slider position at bottom.
+  if (at_bottom)
+    _history_list_view->scrollToBottom ();
 }
 
 void
 history_dock_widget::clear_history (void)
 {
   _history_model->setStringList (QStringList ());
 }
 
